### DISCUSSION TITLE
request headers host unexpectedly changed

### DISCUSSION THREAD

**Discussion by mjkohan** [vercel/next.js#60069] [App Router]
i have a multi-tenancy app. 
the req headers suddenly working unexpected
const hostname = req.headers.get("host")
the host name is [::1]:51709. it was like :"shop1.exmple.com"
im using req.headers.get("referer") for the hotfix but its null sometimes and im looking for a right solution

**Comment by brendanmoore** [comment]
This issue was introduced in https://github.com/vercel/next.js/pull/53131 in this release: https://github.com/vercel/next.js/releases/tag/v13.4.15 downgrading to `13.4.14` or below fixes the issue. It looks like NextJS exposes the `hostname` in other ways internally that would be really helpful, but its not accessible in pages router (so it seems)
